src/ctia/task/migration/migrate_es_stores.clj

```diff
@@ -75,6 +77,109 @@
   (let [coercer (coerce-to-fn Model)]
     #(reduce (partial append-coerced coercer) {} %)))
 
+(s/defschema BatchParams
+  (st/optional-keys
+   {:source-store StoreMap
+    :target-store StoreMap
+    :migrated-count s/Int
+    :buffer-size s/Int
+    :search_after [s/Any]
+    :migrations s/Any
+    :entity-type s/Keyword
+    :batch-size s/Int
+    :migration-id s/Str
+    :list-coerce s/Any
+    :migration-es-conn ESConn
+    :confirm? s/Bool
+    :documents [{s/Any s/Any}]
+    :query ESQuery}))
+
+
+(s/defn read-source :- (s/maybe BatchParams)
+  "This function retrieves in `source-store`a batch of documents that match the given `query`.
+   When not nil, the `search_after` parameter is used to skip previously retrieved data. The
+   returned result prepares the next batch parameters with new `search_after` along with the
+   documents that have to be written in target."
+  [{:keys [source-store
+           search_after
+           batch-size
+           query]
+    :as batch-params} :- (s/maybe BatchParams)]
+    ;; loop that reads batches from source and produces them in chan
+  (when batch-params
+    (let [{:keys [data paging]} (mst/query-fetch-batch query
+                                                       source-store
+                                                       batch-size
+                                                       0
+                                                       "asc"
+                                                       search_after)
+          next-search-after (:sort paging)]
```

Comment:
It's copy / paste inheritance from former code.
I have renamed this funciton in `read-source-batch` and added a `read-source` functon that generates the lazy sequence. This last function is simpler using `sort` which is always returned by clj-momo pagination function. I tried to use `(-> paging :next :search_after)` but when the last batch is smaller than `batch-size`, [`search_after` is nil](https://github.com/threatgrid/clj-momo/blob/master/src/clj_momo/lib/es/pagination.clj#L30), and if I pass `nil` in the next batch it will restarts from the beginning for given query. 


by `(-> paging :next :search_after)`, but it doesn't work, I think that this is a bug here: https://github.com/threatgrid/clj-momo/blob/master/src/clj_momo/lib/es/pagination.clj#L30

Reply:
`:sort` and `:next :search_after` are set with the value of `sort` in `clj-momo`. I do not understand how they can have a different value.

Reply:
here https://github.com/threatgrid/clj-momo/blob/master/src/clj_momo/lib/es/pagination.clj#L45